src/restruct_poc/model/model.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "node.hpp"
+
+#include "src/restruct_poc/solver/constraint.hpp"
+#include "src/restruct_poc/types.hpp"
+
+namespace openturbine {
+
+static Node GroundNode(-1, {0., 0., 0., 1., 0., 0., 0.});
+
+struct Model {
+    std::vector<Node> nodes;
+    std::vector<Constraint> constraints;
+
+    const Node AddNode(
```

Comment:
Should be return a `const Node&` from this? Also, shouldn't a corresponding `GetNode` method do this job?

Reply:
I originally was returning a node reference, and storing it in `BeamNode`, but they become invalid once another node is added to the model. We should discuss ways to make this better and more ergonomic. I'd really like to have the ability to renumber the nodes once they've all been generated and connected to elements/constraints, but I didn't have a change to figure out how to do it yet.